src/parser.ts

```diff
@@ -0,0 +1,113 @@
+import { getAge, getDateOfBirth, getFullBirthYear, splitPersonalNumberValue } from './utils';
+import { Gender } from './model/Gender.enum';
+import { personalNumberAddingTwentyIssueYear, unprobableMonthAddition, womanMonthAddition } from './model/constants';
+import { ParsingResult } from './model/ParsingResult.interface';
+
+/**
+ * Personal number parser. Returns age, date of birth, gender, birth order from that day, and a control digit from given personal number.
+ *
+ * @param value - The personal number with or without a slash symbol.
+ * @returns ParsingResult
+ * If value is a valid personal number, ParsingResult.result will set to parsing result.
+ * If value is not a valid personal number, ParsingResult.result will be set to undefined, ParsingResult.message will contain an error message.
+ */
+export const parse = (value: string): ParsingResult => {
+  if (!value) {
+    return {
+      result: undefined,
+      message: 'Personal number value cannot be empty.'
+    };
+  }
+
+  const { firstPart, secondPart } = splitPersonalNumberValue(value);
+  if (firstPart.length !== 6) {
+    return {
+      result: undefined,
+      message: `First part of personal number must have 6 digits. Given first part: ${firstPart}.`
```

Comment:
Tady a v tom elsu bych vratil i cely ten input. Protože když ve splitPersonalNumberValue bude vycházet délka >2, tak to vrátí ve firstPart i secondPart prázdný string - to uživatele dost zmate